src/string/builder.cr

```diff
@@ -61,6 +61,29 @@ class String::Builder
     @bytesize == 0
   end
 
+  # Chomps the last byte from the string buffer.
+  # If the byte is '\n' and there's a '\r' before it, it is
+  # also removed.
+  def chomp!(byte : UInt8)
+    if bytesize > 0 && buffer[bytesize - 1] == byte
+      back(1)
+
+      if byte === '\n' && bytesize > 0 && buffer[bytesize - 1] === '\r'
```

Comment:
At least Ruby always removes `\r` when chomping `\n` or the argless version:

```
$ irb
irb(main):001:0> "foo\r\n".chomp("\n")
=> "foo"
irb(main):002:0> "foo\r\n".chomp
=> "foo"
```

So I think it's fine.